File: root/Assets/NakedGame/NakedScripts/ShootNew.js

var projectile : Rigidbody;
var speed = 0.5;
var ammo : int = 0;
var maxAmmo : int = 10;
var ammoText : GUIText;




function Update ()
{	
	if(Input.GetButtonUp ("Fire1") && ammo < maxAmmo)
	{
	
		clone = Instantiate(projectile, transform.position, transform.rotation);
		clone.velocity = transform.TransformDirection(Vector3 (0, 0, speed));
		ammo++;
		
	Destroy (clone.gameObject, 5);
	GetComponent.<ParticleSystem>().Play();
	}
	if( ammo == maxAmmo ) {
	Application.LoadLevel("NakedEnd");
	}
}

function OnGUI()
{
	ammoText.text = "Ammo: " + ammo + "/" + maxAmmo;
	}

File: root/Assets/NakedGame/NakedScripts/ShootNew.cs
using UnityEngine;
using System.Collections;

public class ShootNew : MonoBehaviour
{
    public GameObject projectile;
    public float speed = 25f;
    private int ammo = 10;
    public float maxAmmo = 10f;
    public GUIText ammoText;
	public Timer _timer;
 
    // Update is called once per frame
    void Update()
    {
        if (Input.GetButtonUp("Fire1") && ammo > 0)
        {

            GameObject clone = Instantiate(projectile, transform.position, transform.rotation) as GameObject;
			clone.transform.SetParent(gameObject.transform);


			clone.GetComponent<Rigidbody>().AddForce(0,20,speed);
			clone.transform.SetParent(null);

           //clone.transform = transform.TransformDirection(0, 0, speed);
            ammo --;


            Destroy(clone.gameObject, 5);
			ParticleSystem particle = GetComponent<ParticleSystem>();
			particle.Play();
            
            particle.enableEmission = true;
			ammoText.text = "Ammo: " + ammo + "/" + maxAmmo;
		
		}

		if (ammo == 0 || _timer.timerDone == true )
		{
			Application.LoadLevel("NakedEnd");
		}
		
	}
	
}